From: Dan Williams <dcbw@redhat.com>
Date: Wed, 3 Sep 2014 17:43:25 -0500
Subject: tui: fix updating of NmtPasswordFields passwords (bgo #733002)

The actual entry is a sub-widget, and was getting updated when the
user changed the password, but those changes were not being
propagated to the NmtPasswordFields object's 'password' property.

https://bugzilla.gnome.org/show_bug.cgi?id=733002

(cherry picked from commit 82b0ea87075144650979d344f34c09c7111c4a4e)
---
 tui/nmt-password-fields.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/tui/nmt-password-fields.c b/tui/nmt-password-fields.c
index ec8b521..be75b74 100644
--- a/tui/nmt-password-fields.c
+++ b/tui/nmt-password-fields.c
@@ -156,6 +156,10 @@ nmt_password_fields_constructed (GObject *object)
 	} else
 		g_clear_object (&priv->show_password);
 
+	g_object_bind_property (priv->entry, "text",
+	                        object, "password",
+	                        G_BINDING_BIDIRECTIONAL | G_BINDING_SYNC_CREATE);
+
 	G_OBJECT_CLASS (nmt_password_fields_parent_class)->constructed (object);
 }
 
